src/vocabularyChecker/detectWrongSyntax.ts

```diff
@@ -0,0 +1,100 @@
+import json from "./AllowedVocab.json";
+/**
+ * Interface containing @param fromIndex: number, @param toIndex: number, @param message: string
+ */
+import Issue from "./Issue";
+/**
+ * Regular expression that looks for any and all words. A word is defined as some sequence of
+ * characters (\w*S) betweeen wordbarriers (\b)
+ */
+const anyWord = new RegExp(/\b(\w*\S)\b/g);
+/**
+ * Predefined dictionary containing all allowed Words
+ */
+const allowedWords = json;
+interface Position { fromIndex: number; toIndex: number; }
+
+/**
+ * @param text
+ * @param invalidWords
+ * Returns a map of invalid words (consisted in @param invalidWords) and their corresponding
+ * positions (starting position and end position) inside of a string.
+ */
+
+export function createMapOfInvalidWords(text: string, invalidWords: string[]): Map<string, [number, number]> {
+	const positions = new Map<string, [number, number]>();
+	for (const word of invalidWords) {
+		positions.set(word, [text.indexOf(word), text.indexOf(word) + word.length - 1]);
+	}
+	return positions;
+}
+
+/**
+ * @param text
+ * @param invalidWords
+ * Returns message that tells the user which words are not allowed and on which
+ * position those are within @param text.
+ */
+export function createErrorMessages(text: string, invalidWords: string[]): string|undefined {
+	const wrongWords = logInvalidWords(createMapOfInvalidWords(text, invalidWords));
+	return wrongWords;
+}
+
+/**
+ * @param text
+ * Just a function to bundle the functionalities of @function collectAllInvalidWords and
+ * @function removeDuplicates.
+ * @returns a Stringarray containing exactly one copy of each wrong word in @param text.
+ */
+export function preprocessText(text: string): string[] {
+	return removeDuplicates(collectAllInvalidWords(text));
+}
+
+/**
+ * @param text
+ * @returns a Stringarray containing every occurrence of each wrong word in @param text.
+ */
+export function collectAllInvalidWords(text: string): string[] {
+	const temp = text.match(anyWord) || [];
+	const result: string[] = [];
+	temp.forEach(element => {
+		if (!allowedWords.includes(element)) {
+		result.push(element);
+		}
+	});
+ return result;
+}
+
+/**
+ * @param invalidWords
+ * @returns a Stringarray containing the same elements as @param invalidWords
+ * minus duplicate elements.
+ */
+function removeDuplicates(invalidWords: string[]): string[] {
+	return invalidWords.filter((value, item) => invalidWords.indexOf(value) === item);
```

Comment:
You can simplify and generalize this (it works for any type `T`, not just `string`, and `Set` is a type that doesn't contain duplicates):
```
function removeDuplicates<T>(values: T[]): T[] {
  return [...new Set(values)];
}
```

Reply:
If I understand you correctly I could even remove this whole function, if I just stick to using Set from the start, right?

Reply:
If sets are all you need then yes, go for it. 

Don't overestimate sets in general though; they use `!(a !== b)` to compare elements stored within them, *not a deep comparison*. Keep that in mind before storing objects or arrays in them.

If you decide to go with @Nifdee s suggestion for a utility method, I would implement it as 
```typescript
export function uniqueValues<T>(values: Iterable<T>): Iterable<T> {
  return new Set(values);
}
```
and put it into it's own file (or a file that contains all the `Iterable`-functions we'll come up with.

Reply:
For now, removeDuplicates works appropriately. I will see how it works with Sets, though.